dockerfiles/feature-support/hsm-patches/0001-Introduce-generic-keypair-generation-interface-and-e.patch

```diff
@@ -424,12 +423,11 @@ index ec7f279..5a30e79 100644
 +	int curve_nid = NID_undef;
 +
 +	// R/W session is mandatory for key generation.
-+	if (pkcs11_open_session(slot, 1)) {
-+		return -1;
-+	}
-+	// open_session might call C_CloseAllSessions if current session is not R/W.
-+	// C_CloseAllSessions logs everyone out
-+	if (slot->logged_in <= 0) {
++	if (slot->rw_mode == 0) {
```

Comment:
I think there is a race condition here

Reply:
Also, can't rw also be -1? https://github.com/OpenSC/libp11/blob/281254bf3883ed17f4756243d78ac906342211d8/src/p11_slot.c#L427

Reply:
Discussed offline on thread safety guarantees (or lack there of by MoCOCrW).
See:
https://github.com/bmwcarit/MoCOCrW/blob/cc2acba5d6e55c94a144cb9306c9fb2345153911/README.md?plain=1#LL84C48-L84C90